swift-sdk/Internal/Auth.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Created by Jay Kim on 6/25/20.
+//  Copyright © 2020 Iterable. All rights reserved.
+//
+
+import Foundation
+
+protocol AuthProvider: AnyObject {
+    var auth: Auth { get }
+}
+
+struct Auth {
+    let userId: String?
+    let email: String?
+    var authToken: String?
```

Comment:
change to `let`. Unless there is a good reason, we should not create mutable structs.